modules.d/03systemd-hostnamed/module-setup.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+# called by dracut
+check() {
+    [[ $mount_needs ]] && return 1
+
+    if ! dracut_module_included "systemd"; then
+        derror "systemd-hostnamed needs systemd in the initramfs"
+        return 1
+    fi
+
+    return 0
+}
+
+# called by dracut
+depends() {
+    echo "systemd systemd-networkd"
+}
+
+installkernel() {
+    return 0
+}
+
+# called by dracut
+install() {
+    local _mods
+
+    inst_multiple -o \
+       $systemdutildir/systemd-hostnamed \
+       $systemdsystemunitdir/systemd-hostnamed.service \
+       hostname hostnamectl
+
+    if [[ $hostonly ]]; then
+       inst_multiple -H -o \
+           /etc/hosts \
+           /etc/hostname \
+    	   /etc/machine-info \
+           ${NULL}
+    fi
+
+    _arch=${DRACUT_ARCH:-$(uname -m)}
+    inst_libdir_file {"tls/$_arch/",tls/,"$_arch/",}"libnss_myhostname.so.*"
+
+    systemctl -q --root "$initdir" enable systemd-hostnamed
+
+}
```

Comment:
Please also add the module in the `network` section of `dracut.spec`.

Reply:
@bengal it's up to the package maintainers to add or remove any relevant modules for their distro and this work is far from completed. On top of that the dracut spec file should arguably be split into three separated files ( Fedora/RHEL/SUSE )

Reply:
> @bengal it's up to the package maintainers to add or remove any relevant modules for their distro and this work is far from completed. 

Fair enough.

> On top of that the dracut spec file should arguably be split into three separated files ( Fedora/RHEL/SUSE )

As far as I understand, the spec file already supports those 3 distros. A single file with conditionals (as it is now) seems better than having to maintain 3 files with a lot of duplication.

Reply:
> As far as I understand, the spec file already supports those 3 distros. A single file with conditionals (as it is now) seems better than having to maintain 3 files with a lot of duplication.

The only pros you have by maintaining multiple distribution in a single rpm spec file is if it's being built by the same buildsystem and used by the same test system and test cases which may or may not be the case here since I have heard RH sales person claiming that RH build and maintained majority of Suse packages ( which was more likely to have been a sales pitch to get company to chose RHEL over Suse enterprise ) but since I dont work for either of those company's I cant say if that's really the case or not.

It's unmaintainable for upstream in general trying to support all the what ca 50 packaging solution in the fragmented downstream linux distribution ecosystem which is why I personally feel that dracut should just support/ship packaging example ( and tests ) for the primary's as in Arch,Debian,Fedora,Gentoo and Opensuse unless downstream provides resources to support/maintain those files upstream

On top of that if I was a downstream package maintainer I would ensure that each of the dracut modules would be sub package of the component they represent since it's the only way to keep the package dependency chain intact and installs small which is not the case in the dracut spec file.